File: core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/promisehandlers/OpenExecutorHandler.scala
package edu.uci.ics.amber.engine.architecture.worker.promisehandlers

import edu.uci.ics.amber.engine.architecture.worker.DataProcessorRPCHandlerInitializer
import edu.uci.ics.amber.engine.architecture.worker.promisehandlers.OpenExecutorHandler.OpenExecutor
import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.ControlCommand

trait OpenExecutorHandler {
  this: DataProcessorRPCHandlerInitializer =>
  registerHandler { (_: OpenExecutor, sender) =>
    dp.executor.open()
  }
}

File: core/amber/src/main/python/core/architecture/handlers/control/open_executor_handler.py
from core.architecture.handlers.control.control_handler_base import ControlHandler
from proto.edu.uci.ics.amber.engine.architecture.rpc import EmptyReturn


class OpenExecutorHandler(ControlHandler):

    def open_executor(self) -> EmptyReturn:
        self.context.executor_manager.executor.open()
        return EmptyReturn()
